nextjs/ssr/pages/index.tsx

```diff
@@ -0,0 +1,144 @@
+import { SuperJSONResult } from "superjson/dist/types";
+
+const ssrInput: KitchenSinkRequest = new KitchenSinkRequest({
+  requestDouble: 1.1,
+  requestFloat: 2.2,
+  requestInt32: 3,
+  requestInt64: 4n,
+  requestUint32: 5,
+  requestUint64: 6n,
+  requestSint32: 7,
+  requestSint64: 8n,
+  requestFixed32: 9,
+  requestFixed64: 10n,
+  requestSfixed32: 11,
+  requestSfixed64: 12n,
+  requestBool: true,
+  requestString: 'some string',
+  requestBytes: new Uint8Array([1, 2, 3]),
+  requestMapStringBool: {
+    'some key': true,
+  },
+  requestRepeatedDouble: [1.1],
+  requestRepeatedFloat: [2.2],
+  requestRepeatedInt32: [3],
+  requestRepeatedInt64: [4n],
+  requestRepeatedUint32: [5],
+  requestRepeatedUint64: [6n],
+  requestRepeatedSint32: [7],
+  requestRepeatedSint64: [8n],
+  requestRepeatedFixed32: [9],
+  requestRepeatedFixed64: [10n],
+  requestRepeatedSfixed32: [11],
+  requestRepeatedSfixed64: [12n],
+  requestRepeatedBool: [true],
+  requestRepeatedString: ['some string'],
+  requestRepeatedBytes: [new Uint8Array([1, 2, 3])],
+});
+
+export const getServerSideProps = async () => {
+  const client = createClient(
+    KitchenSinkService,
+    wrapFetch("calling from getServerSideProps", fetch)
+  );
+  const {
+    responseDouble,
+    responseFloat,
+    responseInt32,
+    responseInt64,
+    responseUint32,
+    responseUint64,
+    responseSint32,
+    responseSint64,
+    responseFixed32,
+    responseFixed64,
+    responseSfixed32,
+    responseSfixed64,
+    responseBool,
+    responseString,
+    responseBytes,
+    responseMapStringBool,
+    responseRepeatedDouble,
+    responseRepeatedFloat,
+    responseRepeatedInt32,
+    responseRepeatedInt64,
+    responseRepeatedUint32,
+    responseRepeatedUint64,
+    responseRepeatedSint32,
+    responseRepeatedSint64,
+    responseRepeatedFixed32,
+    responseRepeatedFixed64,
+    responseRepeatedSfixed32,
+    responseRepeatedSfixed64,
+    responseRepeatedBool,
+    responseRepeatedString,
+    responseRepeatedBytes,
+  } = await client.getKitchenSink(ssrInput);
+
+  const kitchenSinkResponse = {
+    responseDouble,
+    responseFloat,
+    responseInt32,
+    responseInt64,
+    responseUint32,
+    responseUint64,
+    responseSint32,
+    responseSint64,
+    responseFixed32,
+    responseFixed64,
+    responseSfixed32,
+    responseSfixed64,
+    responseBool,
+    responseString,
+    responseBytes,
+    responseMapStringBool,
+    responseRepeatedDouble,
+    responseRepeatedFloat,
+    responseRepeatedInt32,
+    responseRepeatedInt64,
+    responseRepeatedUint32,
+    responseRepeatedUint64,
+    responseRepeatedSint32,
+    responseRepeatedSint64,
+    responseRepeatedFixed32,
+    responseRepeatedFixed64,
+    responseRepeatedSfixed32,
+    responseRepeatedSfixed64,
+    responseRepeatedBool,
+    responseRepeatedString,
+    responseRepeatedBytes,
+  };
+  
+  return {
+    props: {
+      superjsonResult: SuperJSON.serialize(kitchenSinkResponse),
```

Comment:
Couldn't you simply convert to JSON here...

```suggestion
      superjsonResult: kitchenSinkResponse.toJson(),
```

Reply:
yes! see 1bbd57733081b8b17162c47a4767ba4b4469a04d